### PR TITLE
Clean up go modules

### PR DESCRIPTION
**Pull request by Jeffwan**
This is a follow up PR of #123  to clean up go modules after we switch to volcano-api. Removing kubernetes unrelated dependencies help reduce size the repo, especially for users who use `go mod vendor` or `go mod download`

Signed-off-by: Jiaxin Shan <seedjeffwan@gmail.com>

### CODE REVIEW

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Review by shinytang6 — APPROVED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -14,33 +14,13 @@ require (
 	k8s.io/apimachinery v0.16.15
 	k8s.io/client-go v0.16.15
 	k8s.io/code-generator v0.18.3
```
Comment (shinytang6) [inline go.mod:16]: `k8s.io/code-generator` can also be downgraded to v1.16.15 by the way though it has no effect here :)

**Review by gaocegege — COMMENTED** [review]
/lgtm

Clean!

**Comment by gaocegege** [comment]
The command "./hack/verify-codegen.sh" exited with 1.

File: go.mod (hunk 1 above)
Comment (Jeffwan) [reply go.mod:16]: Yeah. I will fix it

**Review by gaocegege — COMMENTED** [review]
/lgtm

**Comment by Jeffwan** [comment]
> The command "./hack/verify-codegen.sh" exited with 1.

fixed.

**Pull request merged**